transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/CustomAttributesStoreImpl.kt

```diff
@@ -7,12 +7,22 @@ import com.genesys.cloud.messenger.transport.util.logs.LogMessages
 import io.ktor.utils.io.charsets.Charsets
 import io.ktor.utils.io.core.toByteArray
 
+private const val MAX_CUSTOM_DATA_BYTES_UNSET = -1
+
 internal class CustomAttributesStoreImpl(
     private val log: Log,
     private val eventHandler: EventHandler,
 ) : CustomAttributesStore {
     private var customAttributes: MutableMap<String, String> = mutableMapOf()
-    var maxCustomDataBytes: Int = 0
+    var maxCustomDataBytes: Int = MAX_CUSTOM_DATA_BYTES_UNSET
+        internal set(value) {
+            if (!value.isUnset()) {
```

Comment:
Something does not properly work in this `if` condition. 

1.  `maybeReportFailure` relies on the value of maxCustomDataBytes when size is validated. But the maxCustomDataBytes field is set only in line 24 (field = value). This leads to situation where isSizeValid return false while it is not correct.